tests/workbenches/notebook-controller/test_spawning.py

```diff
@@ -0,0 +1,201 @@
+#
+# Copyright OpenDataHub authors.
+# License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+#
+from __future__ import annotations
+
+import functools
+import io
+import logging
+import pathlib
+
+import allure
+import ocp_resources.namespace
+import ocp_resources.persistent_volume_claim
+import ocp_resources.route
+import ocp_resources.resource
+
+import kubernetes.client
+from kubernetes.dynamic import DynamicClient
+
+from tests.workbenches.conftest import OdhAnnotationsLabels, OdhConstants, PodUtils
+from tests.workbenches.docs import TestDoc, SuiteDoc, Contact, Desc, Step
+
+
+# @allure.title("Test Authentication")
+# @allure.description("Verifies deployments of Notebooks via GitOps approach")
+# @allure.tag("NewUI", "Essentials", "Authentication")
+# @allure.severity(allure.severity_level.CRITICAL)
+# @allure.label("owner", "John Doe")
+# @allure.link("https://dev.example.com/", name="Website")
+# @allure.issue("AUTH-123")
+# @allure.testcase("TMS-456")
+
+
+@SuiteDoc(
+    description=Desc("Verifies deployments of Notebooks via GitOps approach"),
+    beforeTestSteps={
+        Step(value="Deploy Pipelines Operator", expected="Pipelines operator is available on the cluster"),
+        Step(value="Deploy ServiceMesh Operator", expected="ServiceMesh operator is available on the cluster"),
+        Step(value="Deploy Serverless Operator", expected="Serverless operator is available on the cluster"),
+        Step(value="Install ODH operator", expected="Operator is up and running and is able to serve it's operands"),
+        Step(value="Deploy DSCI", expected="DSCI is created and ready"),
+        Step(value="Deploy DSC", expected="DSC is created and ready"),
+    },
+    afterTestSteps={
+        Step(
+            value="Delete ODH operator and all created resources",
+            expected="Operator is removed and all other resources as well",
+        )
+    },
```

Comment:
I don't think this has been functionally implemented but FWIW we don't plan to remove the operator after the test run

Reply:
fixed the doc comment, I never tried to implement operator removal here